File: frontend/src/types/stats.ts
export interface MetricCard {
  title: string;
  value: string;
  description: string;
  trend: number;
  trend_type: 'up' | 'down' | 'neutral';
  iconName: 'dollar' | 'users' | 'file-text' | 'trending-up';
}

export interface ChartDataPoint {
  label: string;
  revenue: number;   
  collected: number; 
}

export interface DistributionPoint {
  name: string;
  value: number;
}

export interface PendingPayment {
  id: number;
  client_name: string;
  prime_total: number;
  solde: number;
  mode_paiement: string;
}

export interface DashboardStatsResponse {
  cards: MetricCard[];
  main_chart: ChartDataPoint[];
  distribution_chart: DistributionPoint[];
  pending_payments: PendingPayment[];
}

File: backend/app/schemas/stats.py
from pydantic import BaseModel
from typing import List, Optional

class MetricCard(BaseModel):
    title: str
    value: str
    trend: float
    trend_type: str
    description: str
    iconName: str 

class ChartDataPoint(BaseModel):
    label: str
    revenue: float
    collected: float

class DistributionPoint(BaseModel):
    name: str
    value: int
    
class PendingPayment(BaseModel):
    id: int
    client_name: str
    prime_total: float
    solde: float
    mode_paiement: Optional[str] = "N/A"

class DashboardStatsResponse(BaseModel):
    cards: List[MetricCard]
    main_chart: List[ChartDataPoint]
    distribution_chart: List[DistributionPoint]
    pending_payments: List[PendingPayment]